core/src/main/java/org/pentaho/di/connections/vfs/BaseVFSConnectionProvider.java

```diff
@@ -90,4 +103,73 @@ protected static boolean getBooleanValueOfVariable( VariableSpace space, String
   protected VariableSpace getSpace( ConnectionDetails connectionDetails ) {
     return connectionDetails.getSpace() == null ? Variables.getADefaultVariableSpace() : connectionDetails.getSpace();
   }
+
+  @Override
+  public boolean test( T connectionDetails, VFSConnectionTestOptions connectionTestOptions ) throws KettleException {
```

Comment:
Could you please use the `@NonNull` and `@Nullable` annotations as suggested in https://github.com/pentaho/pentaho-kettle/pull/9305#discussion_r1595496535?

Reply:
addressed